### PR TITLE
Changes for ATS4

### PR DESCRIPTION
**Pull request by rbberger**
## PR Summary

- adds CI for ATS4
- updates patch for spark-variant to support ROCm

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file.
- [ ] LANL employees: make sure tests pass both on the github CI and on the Darwin CI

If preparing for a new release, in addition please check the following:
- [ ] Update the version in cmake.
- [ ] Move the changes in the CHANGELOG.md file under a new header for the new release, and reset the categories.
- [ ] Ensure that any `when='@main'` dependencies are updated to the release version in the package.py

### CODE REVIEW

File: .gitlab/build_and_test.cmake
Comment (Yurlungur) [inline .gitlab/build_and_test.cmake:1]: Nice. This is a good addition.

File: .gitlab/common.yml
Comment (Yurlungur) [inline .gitlab/common.yml:1]: :+1: for splitting this out

**Review by Yurlungur — APPROVED** [review]
Awesome to have this in. Please update Kokkos in the submodule and the spackage too if needed.

**Comment by rbberger** [comment]
I still want to update this next week to use the Cray Fortran compiler to enable building our Fortran interfaces. I already tested this, I just need to propagate this to the CI. We do also have an issue of very slow build times with -O1 or higher.

**Pull request merged**